src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -657,134 +657,180 @@ public void getDocIds(DocIdPusher pusher) throws InterruptedException {
     if (!this.markAllDocsAsPublic) {
       MemberService memberService =
           this.soapFactory.newMemberService(documentManagement);
-      Map<GroupPrincipal, List<Principal>> groupDefinitions =
-          getGroups(memberService);
-      if (this.publicAccessGroupEnabled) {
-        List<Principal> publicAccessGroup = getPublicAccessGroup(memberService);
-        if (publicAccessGroup.size() > 0) {
-          groupDefinitions.put(
-              new GroupPrincipal("[Public Access]", this.localNamespace),
-              publicAccessGroup);
+      DocIdPusher.FeedType feedType = DocIdPusher.FeedType.FULL;
+      SOAPFaultException groupsException = null;
+      Groups groups = new Groups(memberService, this);
+      try {
+        groups.addGroups();
+        if (this.publicAccessGroupEnabled) {
+          groups.addPublicAccessGroup("[Public Access]", this.localNamespace);
         }
+        if (!groups.isFull()) {
+          feedType = DocIdPusher.FeedType.INCREMENTAL;
+        }
+      } catch (SOAPFaultException e) {
+        groupsException = e;
+        feedType = DocIdPusher.FeedType.INCREMENTAL;
```

Comment:
Why change  feedType?

Reply:
If an exception was thrown, we probably don't have the full feed, but we still might have some information, so I thought in that case we wanted to send what we had without deleting existing data.